docs/breaking.md

```diff
@@ -1,5 +1,406 @@
+
+There are no other endpoints marked for removal to share at this time.
+
+### Consistent HTTP Responses
+
+We've caught a few situations where some API endpoints don't respond with the expected HTTP Status code. We will be standardizing our HTTP responses to make sure that all our `v2` endpoints provide a consistent HTTP response status.
+
+#### Successful Status Codes
+
+200 OK - responds with a custom payload
+201 Created - responds with the created entity
+202 Accepted -  responds with `{“message”: “Accepted.”}`
+
+#### Error Status Codes
+
+##### User Errors
+
+For user fault errors, the error message will be human readable and have enough information for a developer to understand and address the problem.
+
+###### 404 Not Found
+
+Generally responds with:
+
+```json
+{“message”: “some-entity-name not found.”}
+```
+
+This code will be used when:
+
+- The requested resource is not found
+- The URI is incorrect, including project slug being incorrect, or job number/workflow ID being malformed. This does not include query parameters, which we can parse and validate.
+- Unauthenticated requests to resource
+
+###### 400 Bad Request
+
+Generally responds with: 
+
+```json
+{
+    “errors”: [{“type”:  “error-type”,
+                “value”: “some value”}...]
+}
+```
+
+Use this error code when:
+
+- Query parameters are missing or invalid
+- Request body is invalid
+
+#### Server Fault Codes
+
+The relevant HTTP 5xx status codes will be used in the situation where a server error occurs. The return body will contain a message explaining the error. For example:
+
+- 500 Internal Server Error - responds with `{“message”: “Internal server error.”}`
+- 503 Service Unavailable - responds with `{“message”: “Service unavailable.”}`
+
+### Changes to Routes
+
+- [Get Test Metadata](https://circleci.com/docs/api/v2/#get-test-metadata) endpoint will move from `/project/{project-slug}/{job-number}/tests` to `/project/{project-slug}/job/{job-number}/tests`. Note the addition of `job/` to the route
+- References to `project/` will be pluralized to `projectS/` e.g.
+  - `/project/{project-slug}` -> `/projects/{project-slug}`
+  - `/project/{project-slug}/job/{job-number}` -> `/projects/{project-slug}/job/{job-number}`
+- References to `pipeline/` will be pluralized to `pipelineS/` e.g.
+  - `/pipeline/{pipeline-id}` -> `pipelineS/{pipeline-id}`
+  - `/pipeline/{pipeline-id}/config` -> `/pipelineS/{pipeline-id}/config`
+
+### Paginated Responses
+
+- [Get an environment variable](https://circleci.com/docs/api/v2/#get-an-environment-variable) response will be paginated
+- [Get all checkout keys](https://circleci.com/docs/api/v2/#get-all-checkout-keys) response will be paginate
+
+### Representation of Objects in Responses
+
+We're also proposing a change the shape of how objects will be represented in our responses:
```

Comment:
I'm not sure it'll be helpful for people to read this because it might be hard to connect to changes in responses for particular endpoints. If we really want to be upfront and clear about what we're going to change it might be better to give a list of changed endpoints with proposed responses.